storage/innobase/fil/fil0fil.cc

```diff
@@ -2084,6 +2093,22 @@ fil_rename_tablespace_check(
 
 	exists = false;
 	if (os_file_status(new_path, &exists, &ftype) && !exists) {
+		auto schema_path= fil_make_filepath(new_path, NULL, NO_EXT,
+						    true, true);
+		if (schema_path == NULL) {
+			return DB_ERROR;
+		}
```

Comment:
We could avoid deepening the indentation by writing this as follows:
```c++
	if (!os_file_status(new_path, &exists, &ftype) || exists) {
	} else if (auto schema_path = fil_make_filepath(/*…*/)) {
		// ...
		ut_free(schema_path);
		return err;
	} else {
		return DB_ERROR;
	}
```

Reply:
Well, perhaps it would lose the readability a bit this way. With a fresh look, I think the check for schema can be done independently and the last updated patch should solve the concern.